### PR TITLE
Add support for Docker

### PR DESCRIPTION
**Pull request by bmamlin**
This adds support for running teleirc via Docker or natively. If you are interested in this, you will want to set up your own repo on [Docker Hub](https://hub.docker.com/) so people can always get the latest version of teleirc running with a single command:
1. If you don't already have an account, sign up for a free account at https://hub.docker.com/ (e.g., RITlug)
2. Link your Docker Hub account to your GitHub account ([instructions](https://docs.docker.com/docker-hub/builds/))
3. Change the image reference in the `docker run` example in README.md from "burke/teleirc" to match your Docker Hub account (e.g., "RITlug/teleirc")
4. Merge this pull request
5. Log into Docker Hub and select Create > Create Automated Build in the top right to define a public "teleirc" image
6. Under [teleirc github settings](https://github.com/RITlug/teleirc/settings), go to Integrations & services and add "Docker" service to trigger automatic builds of the Docker Hub image when commits are made to github.

Once that's done, anyone can run teleirc within a single docker run command by passing in their configuration via environment variables. 😃 

On the other hand, if you don't want to support Docker in this way, just close this PR and I'll understand.

### CODE REVIEW

File: config.js.docker [hunk 1]
```diff
@@ -0,0 +1,22 @@
+var settings = {
```
Comment (thenaterhood) [inline config.js.docker]: Would there be any reason not to do this for the default config? I feel like looking in process variables is a solid way to go for things. @jflory7 @repkam09

File: README.md [hunk 2]
```diff
@@ -64,13 +98,14 @@ To get teleirc working, you will need a server to run it on, git, and the latest
 
 Alternatively, if you start up the bot with no Telegram chat ID set, it will sit waiting for messages to be sent to it. If you invite the bot to your group chat, you should see a "Debug TG" message with some information about the invite that was sent. One of the fields here will be the chatId. This is the value that needs to be put in the config object. Be careful not to get the user ID of a specific user when reading these messages.
 
-#### Example: config.js
+##### Example: config.js
 
 ```javascript
-{
+var settings = {
```
Comment (robbyoconnor) [inline README.md]: use `let`

File: config.js.docker (hunk 1 above)
Comment (robbyoconnor) [inline config.js.docker]: Thre is no reason for two configuation files to exist IMHO...

File: README.md [hunk 3]
```diff
@@ -64,13 +98,14 @@ To get teleirc working, you will need a server to run it on, git, and the latest
...
     token: "000000000:AAAAAAaAAa2AaAAaoAAAA-a_aaAAaAaaaAA",
     ircBlacklist: [
-        "CowSayBot"
+        "CowSayBot",
+        "AnotherNickToIgnore"
```
Comment (robbyoconnor) [inline README.md]: Remove `AnotherNickToIgnore`, however keep the trailing comma.

**Comment by jflory7** [comment]
@bmamlin Think you'll have a chance to make any of the above corrections? Hoping to cut the new 1.1.0 release by the end of this week, would be great to get this all merged in. :smile:

**Comment by jflory7** [comment]
Actually, thinking this through, it will require us to set up a RITlug presence on Docker Hub. Since I'm really ready to get 1.1.0 out the door, I'm going to tag this for v1.1.1 and we can wait a little longer on it, if necessary.

**Comment by robbyoconnor** [comment]
@bmamlin you may also need to rebase.

**Comment by itamarjp** [comment]
@bmamlin I would like to see this merged, please fix the conflict with README.md, 

also +1 for a Dockerfile.fedora (FROM fedora)

thank you.

**Comment by robbyoconnor** [comment]
use Alpine Linux as a base -- it's a much, much smaller image!

**Comment by robbyoconnor** [comment]
I've used https://hub.docker.com/r/mhart/alpine-node/ 

Example Dockerfile: https://github.com/openmrs/openmrs-contrib-id/blob/master/Dockerfile

**Comment by robbyoconnor** [comment]
No, Use Alpine Linux as a base -- it's a much, much smaller image!
I've used https://hub.docker.com/r/mhart/alpine-node/

Example Dockerfile: https://github.com/openmrs/openmrs-contrib-id/blob/master/Dockerfile

**Comment by itamarjp** [comment]
I fixed the conflict in README.md in https://github.com/RITlug/teleirc/pull/22

**Comment by itamarjp** [comment]
@robbyoconnor be my guest to submit a Docker.alpine file.

**Comment by robbyoconnor** [comment]
Will do :)

**Comment by jflory7** [comment]
#22 has the required changes for this PR, so we will merge this in via that PR.

**Pull request closed without merging**